From b1a95863c6b8d36a01632301971d1f3a56a00b9a Mon Sep 17 00:00:00 2001
From: Bharat Bhushan <Bharat.Bhushan@freescale.com>
Date: Thu, 26 Nov 2015 20:52:31 +0530
Subject: [PATCH 100/508] vfio fsl-mc: Make vfio irq handler non-threaded

For performance we need to make vfio handler non-threaded.
With this we see a lot more improved performance with most
of networking benchmarks.

Signed-off-by: Bharat Bhushan <Bharat.Bhushan@freescale.com>
[Original patch taken from QorIQ-SDK-V2.0-20160527-yocto]
Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 drivers/vfio/fsl-mc/vfio_fsl_mc_intr.c |    5 ++---
 1 files changed, 2 insertions(+), 3 deletions(-)

diff --git a/drivers/vfio/fsl-mc/vfio_fsl_mc_intr.c b/drivers/vfio/fsl-mc/vfio_fsl_mc_intr.c
index 5bb1c7c..74bf2f7 100644
--- a/drivers/vfio/fsl-mc/vfio_fsl_mc_intr.c
+++ b/drivers/vfio/fsl-mc/vfio_fsl_mc_intr.c
@@ -296,7 +296,7 @@ static int vfio_fsl_mc_disable_irq(struct fsl_mc_device *mc_dev, int irq_num)
 	return error;
 }
 
-static irqreturn_t vfio_threaded_irq_handler(int irq_num, void *arg)
+static irqreturn_t vfio_fsl_mc_irq_handler(int irq_num, void *arg)
 {
 	struct vfio_fsl_mc_device *vdev;
 	int i;
@@ -339,8 +339,7 @@ int vfio_fsl_mc_configure_irq(struct vfio_fsl_mc_device *vdev,
 			    0, mc_irq->name, mc_irq);
 	if (error < 0) {
 		dev_err(&mc_dev->dev,
-			"devm_request_threaded_irq() failed: %d\n",
-			error);
+			"IRQ registration fails with error: %d\n", error);
 		return error;
 	}
 
-- 
1.7.5.4

